test/1.4/lib/T_registers.dml

```diff
@@ -77,34 +78,36 @@ bank array[b < 2] {
         }
     }
 
-    method test() throws {
-        local uint32 val;
+    method test() {
+        try {
+            local uint32 val;
 
-        val = read(0x4, 0xffffffff, NULL);
-        assert val == 0x76543210;
+            val = read(0x4, 0xffffffff, NULL);
+            assert val == 0x76543210;
 
-        val = read(0x0, 0xffffffff, NULL);
-        assert val == (b == 0 ? 0xbadc0ffe : 0xdeadbeef);
-        val = read(0x10, 0xffffffff, NULL);
-        assert val == 0x10;
-        val = read(0x14, 0xffffffff, NULL);
-        assert val == 0x14;
-        val = read(0x18, 0xffffffff, NULL);
-        assert val == 0x18;
-        val = read(0x1c, 0xffffffff, NULL);
-        assert val == 0x1c;
-        val = read(0x40, 0xffffffff, NULL);
-        assert val == 0x40;
-        val = read(0x50, 0xffffffff, NULL);
-        assert val == 0x50;
-        val = read(0x44, 0xffffffff, NULL);
-        assert val == 0x44;
-        val = read(0x48, 0xffffffff, NULL);
-        assert val == 0x48;
-        val = read(0x54, 0xffffffff, NULL);
-        assert val == 0x54;
-        val = read(0x58, 0xffffffff, NULL);
-        assert val == 0x58;
+            val = read(0x0, 0xffffffff, NULL);
+            assert val == (b == 0 ? 0xbadc0ffe : 0xdeadbeef);
+            val = read(0x10, 0xffffffff, NULL);
+            assert val == 0x10;
+            val = read(0x14, 0xffffffff, NULL);
+            assert val == 0x14;
+            val = read(0x18, 0xffffffff, NULL);
+            assert val == 0x18;
+            val = read(0x1c, 0xffffffff, NULL);
+            assert val == 0x1c;
+            val = read(0x40, 0xffffffff, NULL);
+            assert val == 0x40;
+            val = read(0x50, 0xffffffff, NULL);
+            assert val == 0x50;
+            val = read(0x44, 0xffffffff, NULL);
+            assert val == 0x44;
+            val = read(0x48, 0xffffffff, NULL);
+            assert val == 0x48;
+            val = read(0x54, 0xffffffff, NULL);
+            assert val == 0x54;
+            val = read(0x58, 0xffffffff, NULL);
+            assert val == 0x58;
+        } catch assert false;
```

Comment:
This would be more optimal with a try/catch around each read (so you get the exact read that fails), but its not a big deal

Reply:
I think the main purpose of this code is to validate the read values, the assert false thing might be viewed more as a test consistency check.